salt/srv/salt/_modules/ceph.py

```diff
@@ -397,6 +397,37 @@ def get_cluster_object(cluster_name, sync_type, since):
             assert ret == 0
             data['crush'] = json.loads(raw)
 
+            # Inject DNE OSDs into the map to avoid inconsistencies between
```

Comment:
Fair enough, I just wanted to float it, and that seemed easiest to do with code :-)

One kink though is that without this (or without injecting it somewhere), the DNE OSD isn't in `data['osds']`, which means it doesn't appear in the dashboard, i.e. it's not obvious that you actually have a DNE OSD still in the CRUSH map. OTOH, that behaviour/display is consistent with `ceph -w` output (which also doesn't draw your attention to DNE OSDs) so maybe that's OK...?